server/src/main/java/org/apache/iotdb/db/metrics/dropwizard/reporter/IoTDBReporter.java

```diff
@@ -0,0 +1,225 @@
+      this.metricFilter = metricFilter;
+      return this;
+    }
+
+    public IoTDBReporter build() {
+      return new IoTDBReporter(
+          metricRegistry, prefix, metricFilter, executorService, shutdownExecutorOnStop);
+    }
+  }
+
+  @Override
+  public void report(
+      SortedMap<String, Gauge> gauges,
+      SortedMap<String, Counter> counters,
+      SortedMap<String, Histogram> histograms,
+      SortedMap<String, Meter> meters,
+      SortedMap<String, Timer> timers) {
+    for (Map.Entry<String, Gauge> entry : gauges.entrySet()) {
+      sendGauge(entry.getKey(), entry.getValue());
+    }
+    for (Map.Entry<String, Counter> entry : counters.entrySet()) {
+      sendCounter(entry.getKey(), entry.getValue());
+    }
+    for (Map.Entry<String, Histogram> entry : histograms.entrySet()) {
+      sendHistogram(entry.getKey(), entry.getValue());
+    }
+    for (Map.Entry<String, Meter> entry : meters.entrySet()) {
+      sendMeter(entry.getKey(), entry.getValue());
+    }
+    for (Map.Entry<String, Timer> entry : timers.entrySet()) {
+      sendTimer(entry.getKey(), entry.getValue());
+    }
+  }
+
+  private void sendTimer(String name, Timer timer) {
+    writeSnapshotAndCount(
+        prefixed(name), timer.getSnapshot(), timer.getCount(), 1.0D / TimeUnit.SECONDS.toNanos(1L));
+  }
+
+  private void sendMeter(String name, Meter meter) {
+    double value = meter.getCount();
+    updateValue(prefixed(name), metricsUtils.emptyMap(), value);
+  }
+
+  private void sendHistogram(String name, Histogram histogram) {
+    writeSnapshotAndCount(prefixed(name), histogram.getSnapshot(), histogram.getCount(), 1.0);
+  }
+
+  private void sendCounter(String name, Counter counter) {
+    double value = counter.getCount();
+    updateValue(prefixed(name), metricsUtils.emptyMap(), value);
+  }
+
+  private void sendGauge(String name, Gauge gauge) {
+    Object obj = gauge.getValue();
+    double value;
+    if (obj instanceof Number) {
+      value = ((Number) obj).doubleValue();
+      updateValue(prefixed(name), metricsUtils.emptyMap(), value);
+    } else if (obj instanceof Boolean) {
+      value = ((Boolean) obj) ? 1 : 0;
+      updateValue(prefixed(name), metricsUtils.emptyMap(), value);
+    } else {
+      LOGGER.warn("Invalid type for Gauge {}: {}", name, obj.getClass().getName());
+    }
+  }
+
+  private void writeSnapshotAndCount(String name, Snapshot snapshot, long count, double factor) {
+    updateValue(name, metricsUtils.mapOf("quantile", "0.5"), snapshot.getMedian() * factor);
+    updateValue(
+        name, metricsUtils.mapOf("quantile", "0.75"), snapshot.get75thPercentile() * factor);
+    updateValue(
+        name, metricsUtils.mapOf("quantile", "0.95"), snapshot.get95thPercentile() * factor);
+    updateValue(
+        name, metricsUtils.mapOf("quantile", "0.98"), snapshot.get98thPercentile() * factor);
+    updateValue(
+        name, metricsUtils.mapOf("quantile", "0.99"), snapshot.get99thPercentile() * factor);
```

Comment:
Are the metrics like tp99/tp95 calculated from all history data or from  a period of time, say last five minutes?
If from all history data, it will be meaningless.